Document/0x05h-Testing-Platform-Interaction.md

```diff
@@ -1381,6 +1381,28 @@ There are several ways to perform dynamic analysis:
 1. For the actual persistence: Use the techniques described in the data storage chapter.
 2. For reflection-based approaches: Use Xposed to hook into the deserialization methods or add unprocessable information to the serialized objects to see how they are handled (e.g., whether the application crashes or extra information can be extracted by enriching the objects).
 
+### Testing for Tapjacking Vulnerability (MSTG-PLATFORM-9)
+
+#### Overview
+
+Tapjacking is a vulnerability wherein the malicious application leverages screen overlay feature of android to capture user taps, keystrokes. That is because an active screen overlay can listen for taps and intercept any information being passed in the underlying activity.
+
+#### Static analysis
+
+Check the source code for either or implementation of below techniques:
+
+- Override [onFilterTouchEventForSecurity()](https://developer.android.com/reference/android/view/View#onFilterTouchEventForSecurity%28android.view.MotionEvent%29)
+- Set [android:filterTouchesWhenObscured](https://developer.android.com/reference/android/view/View.html#setFilterTouchesWhenObscured%28boolean%29) to true
+- [FLAG_WINDOW_IS_OBSCURED](https://developer.android.com/reference/android/view/MotionEvent.html#FLAG_WINDOW_IS_OBSCURED)
+
+Presence of any of this technique will safeguard the application from this vulnerability.
+
+#### Dynamic analysis
+
+To identify the existence of this vulnerability, create a sample apk consiting of a single activity and toast dialog. Upon installing the sample apk on the device and post launching it if the toast dialog overlays on the application to test then it is vulnerable to tapjacking vulnerability.
```

Comment:
```suggestion
To identify the existence of this vulnerability, create a sample APK consisting of a single activity and Toast dialog. Upon installing the sample APK on the device and post launching it, if the toast dialog overlays on the application to test then it is vulnerable to Tapjacking.
```

Reply:
I'm not sure on this part. I've just read it quickly and cannot really follow how to test this. Maybe this could be a bit more explicit or use a real example with screenshots (?)